### PR TITLE
Remove expectedException PHPUnit annotation

### PR DESCRIPTION
**Pull request by MathiasReker**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description? | /
| Type?         | improvement
| Category?     | TE
| BC breaks?    | Does it break backward compatibility? no
| Deprecations? | Does it deprecate an existing feature? no
| Fixed ticket? | /
| How to test?  | /

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/14231)

### CODE REVIEW

File: tests-legacy/Unit/Core/Localization/CLDR/CurrencyTest.php [hunk 1]
```diff
@@ -138,10 +138,11 @@ public function testGetSymbols()
     }
 
     /**
-     * @expectedException \PrestaShop\PrestaShop\Core\Localization\Exception\LocalizationException
      */
     public function testGetSymbolsWithInvalidSymbolType()
     {
+        $this->setExpectedException(\PrestaShop\PrestaShop\Core\Localization\Exception\LocalizationException::class);
```
Comment (PierreRambaud) [inline tests-legacy/Unit/Core/Localization/CLDR/CurrencyTest.php]: Use namespace everywhere please :)

Comment (MathiasReker) [reply tests-legacy/Unit/Core/Localization/CLDR/CurrencyTest.php]: Fixed

File: tests-legacy/Unit/Core/Localization/CLDR/CurrencyTest.php [hunk 2]
```diff
@@ -138,10 +138,11 @@ public function testGetSymbols()
     }
 
     /**
```
Comment (PierreRambaud) [inline tests-legacy/Unit/Core/Localization/CLDR/CurrencyTest.php]: Can you remove the comment when not needed?

File: tests-legacy/Unit/Core/Localization/CLDR/LocaleTest.php [hunk 3]
```diff
@@ -193,10 +194,11 @@ public function testGetNumberSymbolsByNumberingSystem()
      * When asking for the number symbols of this system
      * Then an exception should be raised
      *
```
Comment (PierreRambaud) [inline tests-legacy/Unit/Core/Localization/CLDR/LocaleTest.php]: Remove extra line here please

File: tests-legacy/Unit/Core/Localization/CLDR/LocaleTest.php [hunk 4]
```diff
@@ -224,10 +226,11 @@ public function testGetDecimalPattern()
      * When asking for decimal pattern of this system
      * Then an exception should be raised
      *
```
Comment (PierreRambaud) [inline tests-legacy/Unit/Core/Localization/CLDR/LocaleTest.php]: Remove extra line here please

File: tests-legacy/Unit/Core/Localization/CLDR/LocaleTest.php [hunk 5]
```diff
@@ -255,10 +258,11 @@ public function testGetPercentPattern()
      * When asking for percentage pattern of this system
      * Then an exception should be raised
      *
```
Comment (PierreRambaud) [inline tests-legacy/Unit/Core/Localization/CLDR/LocaleTest.php]: Remove extra line here please

File: tests-legacy/Unit/Core/Localization/CLDR/LocaleTest.php [hunk 6]
```diff
@@ -286,10 +290,11 @@ public function testGetCurrencyPattern()
      * When asking for currency (price formatting) pattern of this system
      * Then an exception should be raised
      *
```
Comment (PierreRambaud) [inline tests-legacy/Unit/Core/Localization/CLDR/LocaleTest.php]: Remove extra line here please

File: tests-legacy/Unit/Core/Localization/Currency/CurrencyTest.php [hunk 7]
```diff
@@ -131,10 +132,11 @@ public function testGetSymbol()
      * When requesting the currency symbol for the said locale code
      * Then an exception should be raised
      *
```
Comment (PierreRambaud) [inline tests-legacy/Unit/Core/Localization/Currency/CurrencyTest.php]: Remove extra line here please

File: tests-legacy/Unit/Core/Localization/Currency/CurrencyTest.php [hunk 8]
```diff
@@ -173,10 +175,11 @@ public function testGetName()
      * When requesting the currency name for the said locale code
      * Then an exception should be raised
      *
```
Comment (PierreRambaud) [inline tests-legacy/Unit/Core/Localization/Currency/CurrencyTest.php]: Remove extra line here please

File: tests-legacy/Unit/Core/Localization/Currency/RepositoryTest.php [hunk 9]
```diff
@@ -147,10 +147,11 @@ public function provideValidCurrencyCodes()
      * When asking the currency repository for the corresponding Currency
      * Then an exception should be raised
      *
```
Comment (PierreRambaud) [inline tests-legacy/Unit/Core/Localization/Currency/RepositoryTest.php]: Remove extra line here please

File: tests-legacy/Unit/Core/Localization/Locale/LocaleTest.php [hunk 10]
```diff
@@ -127,10 +128,11 @@ public function testFormatNumber($number, $expected)
      *
      * For more formatting cases, @see \Tests\Unit\Core\Localization\Number\FormatterTest
      *
```
Comment (PierreRambaud) [inline tests-legacy/Unit/Core/Localization/Locale/LocaleTest.php]: Remove extra line here please

File: tests-legacy/Unit/Core/Localization/Locale/LocaleTest.php [hunk 11]
```diff
@@ -211,10 +213,11 @@ public function provideValidPriceData()
...
      * @dataProvider provideInvalidPriceData
      *
```
Comment (PierreRambaud) [inline tests-legacy/Unit/Core/Localization/Locale/LocaleTest.php]: Remove extra line here please

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
See comments

**Comment by MathiasReker** [comment]
Fixed.

**Review by PierreRambaud — APPROVED** [review]

File: tests/Unit/Adapter/MailTemplate/MailTemplateTwigRendererTest.php [hunk 12]
```diff
@@ -61,11 +61,10 @@ public function testConstructor()
         $this->assertNotNull($generator);
     }
 
-    /**
-     * @expectedException \PrestaShop\PrestaShop\Core\Exception\FileNotFoundException
-     */
     public function testFileNotFound()
     {
+        $this->setExpectedException(\PrestaShop\PrestaShop\Core\Exception\FileNotFoundException::class);
```
Comment (PierreRambaud) [inline tests/Unit/Adapter/MailTemplate/MailTemplateTwigRendererTest.php]: use namespace please :)

File: tests/Unit/Core/Data/AbstractTypedCollectionTest.php [hunk 13]
```diff
@@ -84,12 +81,10 @@ public function testRemoveElement()
         $this->assertFalse($elementRemoved);
     }
 
-    /**
-     * @expectedException \PrestaShop\PrestaShop\Core\Exception\TypeException
-     * @expectedExceptionMessage Invalid element type Tests\Unit\Core\Data\InvalidCollectionTestElement, expected Tests\Unit\Core\Data\CollectionTestElement
-     */
     public function testInvalidRemoveElement()
     {
+        $this->setExpectedException(TypeException::class, 'Invalid element type Tests\\Unit\\Core\\Data\\InvalidCollectionTestElement, expected Tests\\Unit\\Core\\Data\\CollectionTestElement');
```
Comment (PierreRambaud) [inline tests/Unit/Core/Data/AbstractTypedCollectionTest.php]: Not sure double backslaces are needed in this file.

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Forget little things :)

**Comment by MathiasReker** [comment]
Fixed

**Comment by PierreRambaud** [comment]
Test are broken: https://travis-ci.org/PrestaShop/PrestaShop/jobs/557935139#L827

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Tests are ok, Thanks @MathiasReker

**Pull request merged**